components/uid-input/test/uid-input.test.ts

```diff
@@ -146,4 +146,29 @@ describe('UidInput', () => {
     expect(value).to.equal("at");
   });
 
+  it('Should translate the default label when attribute localization is set', async () => {
+    uidInput = await fixture(html`
+      <uid-input localization='{"Default label": "Label par défaut"}'></uid-input>
+    `);
+    const label = uidInput.shadowRoot.querySelector('label');
+    expect(label.textContent).equals("Label par défaut");
+  });
+
+  it('Should keep the default label when attribute localization is invalid', async () => {
+    uidInput = await fixture(html`
+      <uid-input localization='{"Default label" "Label par défaut"}'></uid-input>
+    `);
+    const label = uidInput.shadowRoot.querySelector('label');
+    expect(label.textContent).equals("Default label");
+  });
+
+  it('Should keep the default label when attribute localization does not contain the correct key', async () => {
+    uidInput = await fixture(html`
+      <uid-input localization='{"DefaultLabel": "Label par défaut"}'></uid-input>
+    `);
+    const label = uidInput.shadowRoot.querySelector('label');
+    expect(label.textContent).equals("Default label");
+  });
```

Comment:
Maybe add a test "Should keep the default label when attribute localization contain Empty translation?"

Reply:
So you mean to check if the translation is not empty. Yes, we can add this.

Reply:
yes add the test and the implementation, as it work like this in UID today?